web/src/input_pill.js

```diff
@@ -363,6 +366,23 @@ export function create(opts) {
             compose.update_on_recipient_change();
         });
 
+        // clicking on an user pill should open up the user's profile
+        store.$parent.off("click", ".pill");
```

Comment:
@sahil839 You were right. The popup not opening after navigating on streams page was because the click handler was registering twice. The handler gets registered every time a new pill is created.
I'm not sure how we do the clean up in Zulip codebase but I've added this line to ensure that there's only 1 handler at a time which fixes the problem. Let me know your thoughts.

Reply:
I am not sure whether this is the best solution for this.

The click event attached should ideally be removed when removing the element. But in this case we use a delegated event and not a direct event attached to ".pill" element, and so the click event handler is not removed when the ".pill" element is removed.

Reply:
One other solution can be to instead use a direct event here, but that should be set after rendering the pill.

I would wait for more feedback on this from Tim when he reviews this. You can keep this as it is for now.